starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/MainActivity.java

```diff
@@ -20,16 +22,20 @@ protected void onCreate(Bundle savedInstanceState) {
         DBAssetHelper dbSetup = new DBAssetHelper(MainActivity.this);
         dbSetup.getReadableDatabase();
 
-        //Setup the RecyclerView
-        RecyclerView shoppingListRecyclerView = (RecyclerView) findViewById(R.id.shopping_list_recyclerview);
-
-        ShoppingSQLiteOpenHelper db = ShoppingSQLiteOpenHelper.getInstance(this);
-        List<ShoppingItem> shoppingList = db.getShoppingList();
-
-        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this,LinearLayoutManager.VERTICAL,false);
+        Fragment listFragment = FragmentList.newInstance(null, this);
+        getSupportFragmentManager().beginTransaction().add(R.id.frame_layout,listFragment).commit();
+    }
 
-        shoppingListRecyclerView.setLayoutManager(linearLayoutManager);
-        shoppingListRecyclerView.setAdapter(new ShoppingListAdapter(shoppingList));
+    @Override
+    public void onItemSelected(ShoppingItem selectedItem) {
+        Bundle bundle = new Bundle();
+        bundle.putString("name",selectedItem.getName());
+        bundle.putString("descript",selectedItem.getDescription());
+        bundle.putString("price",selectedItem.getPrice());
+        bundle.putString("type",selectedItem.getType());
```

Comment:
One of the requirements was "Gather the details for the selected item from the database (not passed through the intent)". Technically, you're not passing these values via intent, but it's almost the same thing. This works OK in this case, but what if the item had 100 fields of data? Then you'd want to pass just the item ID here, and make a separate database query from the detail fragment to get the one item you need.